Source/Libraries/CorruptCore/StashKey.cs

```diff
@@ -75,7 +75,7 @@ private void StashKeyConstructor(string key, string parentkey, BlastLayer blastl
             GameName = (string)AllSpec.VanguardSpec?[VSPEC.GAMENAME] ?? "ERROR";
             SyncSettings = (string)AllSpec.VanguardSpec?[VSPEC.SYNCSETTINGS] ?? "";
 
-            this.SelectedDomains = ((string[])AllSpec.UISpec["SELECTEDDOMAINS"]).ToList();
+            this.SelectedDomains = ((string[])AllSpec.UISpec?["SELECTEDDOMAINS"])?.ToList();
```

Comment:
`AllSpec.UISpec` isn't available in the test environment since the UI hasn't started. This means that, with the existing code, a `StashKey` can't be constructed in the test environment.

As a cheap work-around, I've added these null checks to the code. A more "proper" solution is probably manually set `AllSpec.UISpec` in a test setup method to something that will work. I tried something like that but it has some unexpected side-effects.

Reply:
I was able to figure out how to initialize this in the test code, and I reverted the `StashKey.cs` change